sig-scalability/charter.md

```diff
@@ -0,0 +1,86 @@
+* 10% of time on **Performance improvements**:
+   * Actually designing and implementing performance improvements to the system.
+
+
+### Notes about the time allocation split above:
+* the numbers are meant as a reminder that balance is necessary to make sure that we are not completely overwhelmed by debugging, you do NOT need to do per-minute accounting of your time,
+* the split is expected to change as our tools improve and flakiness decreases - we hope to devote less time to fire-fighting and more time to performance improvements in the long term.
+
+## What can we do/require from other SIGs
+
+*By regression below we mean a regression identified by the set of release-blocking scalability/performance tests (as defined by [the sig-release-x.y-blocking dashboards](https://github.com/kubernetes/test-infra/blob/master/testgrid/config/config.yaml)).*
+
+* We require that significant changes (in terms of impact on the whole system, not in terms of line count), such as: update of Go version, update of etcd, major impactful architectural changes etc. may only be merged:
+   * with an explicit approval from a SIG-scalability approver, and
+   * after having passed performance testing on biggest supported clusters (unless the scalability approver explicitly states this is not necessary).
+* We can block a feature from transitioning to beta status if (when turned on) it causes a significant degradation of other components in terms of scalability/performance. (To judge what is "significant" ideally we would want an objective way such as "breaks an SLO" or "causes SLI degradation of more than XX%". But we are not there yet, so initially this will be sig-scalability team's call.)
+* We can block a feature from transitioning to GA status if it cannot be used at scale.
+* We can roll back any PR merged to master if it has been identified as a cause of an SLO regression. The offending PR should only be merged again after it has been shown to pass tests at scale.
+* We can pause the merge queue in case a regression on master has been observed, until a particular PR has been *identified* as cause of the regression *and* the regression has been *mitigated* (see below for what it means).
+* We can require a SIG to introduce a regression-catching benchmark test for a scalability-critical functionality.
+
+"Rules of engagement" for pausing merge queue:
+1. observe a scalability regression on master,
+2. pause the merge queue,
+3. identify the PR which caused the regression
+   1. this can be done by reading code changes, bisecting, running A/B tests, etc,
+   2. we say a PR is identified as the cause when we are reasonably confident that it indeed caused a regression, *even if the mechanism is not 100% understood* - this is because we want to minimize the time of merge queue pause
+4. mitigate the regression; this can potentially be different things, e.g.:
+   1. reverting the PR,
```

Comment:
I think this needs to be handled with care to ensure folks are informed of the tradeoff's associated with a revert.  https://github.com/kubernetes/kubernetes/pull/60891#issuecomment-371240524 is a good example that blindly reverting without conversing it through can be dangerous.